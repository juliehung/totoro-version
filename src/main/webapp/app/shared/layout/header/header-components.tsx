import React from 'react';
import { Translate } from 'react-jhipster';

import { UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem, NavItem, NavLink, NavbarBrand } from 'reactstrap';
import { NavLink as Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import appConfig from 'app/config/constants';

export const NavDropdown = props => (
  <UncontrolledDropdown nav inNavbar id={props.id}>
    <DropdownToggle nav caret className="d-flex align-items-center">
      <FontAwesomeIcon icon={props.icon} />
      <span>{props.name}</span>
    </DropdownToggle>
    <DropdownMenu right style={props.style}>
      {props.children || <DropdownItem disabled>No Resource</DropdownItem>}
    </DropdownMenu>
  </UncontrolledDropdown>
);

export const BrandIcon = props => (
  <div {...props} className="brand-icon">
    <img src="content/images/logo.png" alt="Logo" />
  </div>
);

export const Brand = props => (
  <NavbarBrand tag={Link} to="/" className="brand-logo">
    <BrandIcon />
    <span className="brand-title">
      <Translate contentKey="global.title">Dentall</Translate>
    </span>
  </NavbarBrand>
);

export const Version = () => <div className="navbar-version">{appConfig.VERSION}</div>;

export const Home = props => (
  <NavItem>
    <NavLink tag={Link} to="/" className="d-flex align-items-center">
      <FontAwesomeIcon icon="home" />
      <span>
        <Translate contentKey="global.menu.home">Home</Translate>
      </span>
    </NavLink>
  </NavItem>
);

export const Survey = props => (
  <NavItem>
    <NavLink tag={Link} to="/" className="d-flex align-items-center">
      <FontAwesomeIcon icon="home" />
      <span>
        <Translate contentKey="survey.title">Survey</Translate>
      </span>
    </NavLink>
  </NavItem>
);
