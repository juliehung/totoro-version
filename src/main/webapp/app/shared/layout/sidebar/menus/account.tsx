import React from 'react';
import { NavLink } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NavLink as Link } from 'react-router-dom';
import { Translate, translate } from 'react-jhipster';

const accountMenuItemsAuthenticated = (
  <>
    <NavLink tag={Link} to="/account/settings">
      <FontAwesomeIcon icon="wrench" /> <Translate contentKey="global.menu.account.settings">Settings</Translate>
    </NavLink>
    <NavLink tag={Link} to="/account/password">
      <FontAwesomeIcon icon="clock" /> <Translate contentKey="global.menu.account.password">Password</Translate>
    </NavLink>
    <NavLink tag={Link} to="/logout">
      <FontAwesomeIcon icon="sign-out-alt" /> <Translate contentKey="global.menu.account.logout">Sign out</Translate>
    </NavLink>
  </>
);

const accountMenuItems = (
  <>
    <NavLink id="login-item" tag={Link} to="/login">
      <FontAwesomeIcon icon="sign-in-alt" /> <Translate contentKey="global.menu.account.login">Sign in</Translate>
    </NavLink>
    <NavLink tag={Link} to="/register">
      <FontAwesomeIcon icon="sign-in-alt" /> <Translate contentKey="global.menu.account.register">Register</Translate>
    </NavLink>
  </>
);

export const AccountMenu = ({ isAuthenticated = false }) => (
  <>
    <p style={{ marginTop: '20px', marginBottom: '-3px' }}>{translate('global.menu.account.main')}</p>
    {isAuthenticated ? accountMenuItemsAuthenticated : accountMenuItems}
  </>
);

export default AccountMenu;
