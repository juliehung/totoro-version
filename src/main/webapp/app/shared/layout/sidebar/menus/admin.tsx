import React from 'react';
import { NavLink } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { NavLink as Link } from 'react-router-dom';
import { Translate, translate } from 'react-jhipster';

const adminMenuItems = (
  <>
    <NavLink tag={Link} to="/admin/user-management">
      <FontAwesomeIcon icon="user" /> <Translate contentKey="global.menu.admin.userManagement">User management</Translate>
    </NavLink>
    <NavLink tag={Link} to="/admin/metrics">
      <FontAwesomeIcon icon="tachometer-alt" /> <Translate contentKey="global.menu.admin.metrics">Metrics</Translate>
    </NavLink>
    <NavLink tag={Link} to="/admin/health">
      <FontAwesomeIcon icon="heart" /> <Translate contentKey="global.menu.admin.health">Health</Translate>
    </NavLink>
    <NavLink tag={Link} to="/admin/configuration">
      <FontAwesomeIcon icon="list" /> <Translate contentKey="global.menu.admin.configuration">Configuration</Translate>
    </NavLink>
    <NavLink tag={Link} to="/admin/audits">
      <FontAwesomeIcon icon="bell" /> <Translate contentKey="global.menu.admin.audits">Audits</Translate>
    </NavLink>
    {/* jhipster-needle-add-element-to-admin-menu - JHipster will add entities to the admin menu here */}
    <NavLink tag={Link} to="/admin/logs">
      <FontAwesomeIcon icon="tasks" /> <Translate contentKey="global.menu.admin.logs">Logs</Translate>
    </NavLink>
  </>
);

const swaggerItem = (
  <NavLink tag={Link} to="/admin/docs">
    <FontAwesomeIcon icon="book" /> <Translate contentKey="global.menu.admin.apidocs">API</Translate>
  </NavLink>
);

const databaseItem = (
  <NavLink tag="a" href="./h2-console" target="_tab">
    <FontAwesomeIcon icon="hdd" /> <Translate contentKey="global.menu.admin.database">Database</Translate>
  </NavLink>
);

export const AdminMenu = ({ showSwagger, showDatabase }) => (
  <>
    <p style={{ marginTop: '20px', marginBottom: '-2px' }}>{translate('global.menu.admin.main')}</p>
    {adminMenuItems}
    {showSwagger && swaggerItem}
    {showDatabase && databaseItem}
  </>
);
