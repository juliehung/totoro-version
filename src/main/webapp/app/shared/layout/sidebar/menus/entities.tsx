import React from 'react';
import { translate } from 'react-jhipster';
import { NavDropdown } from '../sidebar-components';

export const EntitiesMenu = props => (
  // tslint:disable-next-line:jsx-self-close
  <NavDropdown icon="th-list" name={translate('global.menu.entities.main')} id="entity-menu">
    {/* jhipster-needle-add-entity-to-menu - JHipster will add entities to the menu here */}
  </NavDropdown>
);
