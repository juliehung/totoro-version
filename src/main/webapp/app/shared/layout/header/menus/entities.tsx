import React from 'react';
import { DropdownItem } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Translate, translate } from 'react-jhipster';
import { NavLink as Link } from 'react-router-dom';
import { NavDropdown } from '../header-components';

export const EntitiesMenu = props => (
  // tslint:disable-next-line:jsx-self-close
  <NavDropdown icon="th-list" name={translate('global.menu.entities.main')} id="entity-menu">
    <DropdownItem tag={Link} to="/entity/patient">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.patient" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/appointment">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.appointment" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/registration">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.registration" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/tag">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.tag" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/questionnaire">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.questionnaire" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/hospital">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.hospital" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/accounting">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.accounting" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/calendar">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.calendar" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-unusal-incident">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiUnusalIncident" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-unusal-content">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiUnusalContent" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/patient-identity">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.patientIdentity" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/calendar-setting">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.calendarSetting" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/tooth">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.tooth" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/ledger">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.ledger" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/procedure">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.procedure" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/procedure-type">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.procedureType" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/tooth">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.tooth" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatment" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment-task">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatmentTask" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment-procedure">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatmentProcedure" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment-plan">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatmentPlan" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment-task">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatmentTask" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/drug">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.drug" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/todo">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.todo" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/prescription">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.prescription" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/treatment-drug">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.treatmentDrug" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/disposal">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.disposal" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/registration-del">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.registrationDel" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/condition-type">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.conditionType" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-icd-9-cm">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiIcd9Cm" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-icd-10-cm">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiIcd10Cm" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-procedure">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiProcedure" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-procedure-type">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiProcedureType" />
    </DropdownItem>
    <DropdownItem tag={Link} to="/entity/nhi-icd-10-pcs">
      <FontAwesomeIcon icon="asterisk" fixedWidth />
      &nbsp;
      <Translate contentKey="global.menu.entities.nhiIcd10Pcs" />
    </DropdownItem>
    {/* jhipster-needle-add-entity-to-menu - JHipster will add entities to the menu here */}
  </NavDropdown>
);
