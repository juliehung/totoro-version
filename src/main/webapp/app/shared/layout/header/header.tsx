import './header.css';

import React from 'react';
import { Translate } from 'react-jhipster';
import { Navbar, Nav, NavbarToggler, NavbarBrand, Collapse } from 'reactstrap';
import LoadingBar from 'react-redux-loading-bar';

import { Brand, Version } from './header-components';
import { LocaleMenu } from './menus';

export interface IHeaderProps {
  isAuthenticated: boolean;
  isAdmin: boolean;
  ribbonEnv: string;
  isInProduction: boolean;
  isSwaggerEnabled: boolean;
  currentLocale: string;
  onLocaleChange: Function;
  isMenuOpen: boolean;
}

export interface IHeaderState {
  menuOpen: boolean;
}

export default class Header extends React.Component<IHeaderProps, IHeaderState> {
  state: IHeaderState = {
    menuOpen: false
  };

  handleLocaleChange = event => {
    this.props.onLocaleChange(event.target.value);
  };

  renderDevRibbon = () =>
    this.props.isInProduction === false ? (
      <div className="ribbon dev">
        <a href="">
          <Translate contentKey={`global.ribbon.${this.props.ribbonEnv}`} />
        </a>
      </div>
    ) : null;

  renderVersion = () => !this.props.isInProduction && <Version />;

  renderBrand = open => {
    if (!open) {
      return (
        <div className="left-shift">
          <Brand />
        </div>
      );
    }
    return <Brand />;
  };

  toggleMenu = () => {
    this.setState({ menuOpen: !this.state.menuOpen });
  };

  render() {
    const { currentLocale, isMenuOpen } = this.props;
    return (
      <div id="app-header">
        <LoadingBar className="loading-bar" />
        <Navbar dark expand="sm" className="jh-navbar">
          <NavbarToggler aria-label="Menu" onClick={this.toggleMenu} />
          {this.renderBrand(isMenuOpen)}
          <Collapse isOpen={this.state.menuOpen} navbar>
            <Nav id="header-tabs" className="ml-auto" navbar>
              <LocaleMenu currentLocale={currentLocale} onClick={this.handleLocaleChange} />
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}
