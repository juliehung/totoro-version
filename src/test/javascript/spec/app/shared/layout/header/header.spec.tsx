import React from 'react';
import { shallow } from 'enzyme';

import sinon from 'sinon';

import LoadingBar from 'react-redux-loading-bar';
import { Navbar, Nav } from 'reactstrap';

import { Home, Brand } from 'app/shared/layout/header/header-components';
import { AdminMenu, EntitiesMenu, AccountMenu, LocaleMenu } from 'app/shared/layout/header/menus';
import Header from 'app/shared/layout/header/header';

describe('Header', () => {
  let mountedWrapper;

  const localeSpy = sinon.spy();

  const devProps = {
    isAuthenticated: true,
    isAdmin: true,
    currentLocale: 'en',
    onLocaleChange: localeSpy,
    ribbonEnv: 'dev',
    isInProduction: false,
    isSwaggerEnabled: true,
    isMenuOpen: false
  };
  const prodProps = {
    ...devProps,
    ribbonEnv: 'prod',
    isInProduction: true,
    isSwaggerEnabled: false
  };
  const userProps = {
    ...prodProps,
    isAdmin: false
  };
  const guestProps = {
    ...prodProps,
    isAdmin: false,
    isAuthenticated: false
  };

  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Header {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  // All tests will go here
  it('Renders a Header component in dev profile with LoadingBar, Navbar, Nav and dev ribbon.', () => {
    const component = wrapper();
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    expect(component.find(LoadingBar).length).toEqual(1);
    const navbar = component.find(Navbar);
    expect(navbar.length).toEqual(1);
    expect(navbar.find(Brand).length).toEqual(1);
    const nav = component.find(Nav);
    expect(nav.length).toEqual(1);
    expect(nav.find(LocaleMenu).length).toEqual(1);
  });

  it('Renders a Header component in prod profile with LoadingBar, Navbar, Nav.', () => {
    const component = wrapper(prodProps);
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    const navbar = component.find(Navbar);
    expect(navbar.length).toEqual(1);
    expect(navbar.find(Brand).length).toEqual(1);
    const nav = component.find(Nav);
    expect(nav.length).toEqual(1);
    expect(nav.find(LocaleMenu).length).toEqual(1);
    const ribbon = component.find('.ribbon.dev');
    expect(ribbon.length).toEqual(0);
  });
});
