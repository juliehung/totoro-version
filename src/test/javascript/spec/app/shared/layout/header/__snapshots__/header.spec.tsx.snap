// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders a Header component in dev profile with LoadingBar, Navbar, Nav and dev ribbon. 1`] = `
<div
  id="app-header"
>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <Navbar
    className="jh-navbar"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <NavbarToggler
      aria-label="Menu"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <div
      className="left-shift"
    >
      <Component />
    </div>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        id="header-tabs"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <Component
          currentLocale="en"
          onClick={[Function]}
        />
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;

exports[`Header Renders a Header component in prod profile with LoadingBar, Navbar, Nav. 1`] = `
<div
  id="app-header"
>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <Navbar
    className="jh-navbar"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <NavbarToggler
      aria-label="Menu"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <div
      className="left-shift"
    >
      <Component />
    </div>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        id="header-tabs"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <Component
          currentLocale="en"
          onClick={[Function]}
        />
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
